### PR TITLE
Psi4

### PR DESCRIPTION
**Pull request by dranasinghe**
### Motivation or Problem
Pis4 is an electronic structure package. I wrote an adapter to parse output files. 

### Description of Changes
new psi4 adapter is added along with unit tests.

### Testing
Unit-tests were added to test parser for opt freq calculations for TS and ground state. 

### Reviewer Tips
I used the Qchem adapter. Psi4 uses numerical hessian for opt-frequency and TS calculations. Therefore, there are a lot of single point calculations and I selected the smallest value for the energy. Print option 3 needed for printing force constants. There are no building scan option in psi4.

### CODE REVIEW

File: arkane/common.py [hunk 1]
```diff
@@ -725,3 +725,16 @@ def clean_dir(base_dir_path: str = '',
...
+    """if frequency had imaginary frequency convert to negative number
+        '635.0i' -> -635.0
+        '500.0' -> 500.0
+    """
+    if d.endswith('i'):
+        freq = float(d[:-1]) * -1
+    else:
+        freq = float(d)
+
+    return freq
```
Comment (kspieks) [inline arkane/common.py:740]: Can you add a new line to the end of this file?

File: arkane/ess/psi4.py [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+                        # Convert from atomic units (Hartree/Bohr_radius^2) to J/m^2
+                        f_array.extend([float(f) * 4.35974417e-18 / 5.291772108e-11 ** 2 for f in
+                                        line.replace('[', '').replace(']', '').split()])
+                    force = np.array(f_array).reshape(n_rows, n_rows)
+                line = f.readline()
+
+        return force
+
+    def load_geometry(self):
+
```
Comment (kspieks) [inline arkane/ess/psi4.py:114]: Could you remove this blank line?

File: arkane/ess/psi4.py [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+                    unscaled_frequencies = frequencies
+                    vibration = HarmonicOscillator(frequencies=(frequencies, "cm^-1"))
+                    # modes.append(vibration)
+                    freq.append(vibration)
+
+                    # Read the next line in the file
+                line = f.readline()
+
+                # Read the next line in the file
```
Comment (kspieks) [inline arkane/ess/psi4.py:236]: Is this comment supposed to be here? What about the `line = f.readline()` on the next line?

File: arkane/ess/psi4.py [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+                rot.append(rotation)
+            else:
+                for i in range(3):
+                    inertia[i] *= (constants.a0 / 1e-10) ** 2
+                    rotation = NonlinearRotor(inertia=(inertia, "amu*angstrom^2"), symmetry=symmetry)
+                    # modes.append(rotation)
+                rot.append(rotation)
+
+        translation = IdealGasTranslation(mass=(sum(mass), "amu"))
+        # modes.append(translation)
```
Comment (kspieks) [inline arkane/ess/psi4.py:259]: Seems like a duplicate?

File: arkane/ess/psi4.py [hunk 5]
```diff
@@ -72,8 +70,8 @@ def get_number_of_atoms(self):
             while line != '' and n_atoms == 0:
                 # Automatically determine the number of atoms
                 if 'Center              X                  Y                   Z               Mass ' in line and n_atoms == 0:
-                    for i in range(2):
-                        line = f.readline()
+                    line = f.readline()
```
Comment (kspieks) [inline arkane/ess/psi4.py:73]: Super minor but to address the LGTM, could you change this to be `_ = f.readline()` to help indicate that you're only interested in using the information from the 2nd line that is read in (in line 74)?

File: arkane/ess/psi4.py [hunk 6]
```diff
@@ -72,8 +70,8 @@ def get_number_of_atoms(self):
             while line != '' and n_atoms == 0:
                 # Automatically determine the number of atoms
                 if 'Center              X                  Y                   Z               Mass ' in line and n_atoms == 0:
```
Comment (kspieks) [inline arkane/ess/psi4.py:72]: Is the `n_atoms` condition needed in both line 70 and line 72?

**Review by kspieks — COMMENTED** [review]
Thanks for the PR! Looks like a great addition. Could you add a test for loading the geometry and/or loading the conformer? Those methods just seemed more involved so it may be nice to have a test use the `.out` files to confirm that they work as expected.

The most recent commit (and the other minor changes) can probably be squashed once Alon reviews it :)

File: arkane/ess/psi4.py (hunk 3 above)
Comment (dranasinghe) [reply arkane/ess/psi4.py:236]: Those are carried over from qchem. I will clean up.

File: arkane/ess/psi4.py (hunk 4 above)
Comment (dranasinghe) [reply arkane/ess/psi4.py:259]: I don't think so. I am doing translations after after rotation.

File: arkane/ess/psi4.py (hunk 6 above)
Comment (dranasinghe) [reply arkane/ess/psi4.py:72]: No. I fixed it.

**Comment by dranasinghe** [comment]
@kspieks Thank you for the feedback. I address all of the issues you raised. I also tested psi4-Arkane wit real example and it worked.

File: arkane/ess/psi4.py (hunk 4 above)
Comment (kspieks) [reply arkane/ess/psi4.py:255]: Oh sorry I just didn't understand why `modes.append(translation)` was commented out here. Same with `modes.append(vibration)` on line 232 above.

**Comment by kspieks** [comment]
Thanks for pushing the changes. I now see that the function for testing `load_conformer`, already calls `load_geoemtry`. When I first looked at it, I didn’t see an explicit test for `load_geometry`, so I thought it wasn’t being tested, but I was wrong.

**Comment by dranasinghe** [comment]
I also include new features for Orca. I made a separate commit.

File: arkane/ess/orca.py [hunk 7]
```diff
@@ -132,29 +169,86 @@ def load_geometry(self):
...
+                    frequencies.extend([float(line.split()[1])])
+
+                else:
+                    raise Exception(f'Frequencies not found in {self.logfile}')
+                unscaled_frequencies = frequencies
+                vibration = HarmonicOscillator(frequencies=(frequencies, "cm^-1"))
+                # modes.append(vibration)
+                freq.append(vibration)
+
+        # get moments of inertia from external rotational modes, given in atomic units        line = f.readline()
```
Comment (kspieks) [inline arkane/ess/orca.py:227]: Can you remove `line = f.readline()` that is commented out

File: arkane/ess/orca.py [hunk 8]
```diff
@@ -132,29 +169,86 @@ def load_geometry(self):
...
+                for line in log[(i + 2):]:
+                    if not line.strip():
+                        break
+                    frequencies.extend([float(line.split()[1])])
+
+                else:
+                    raise Exception(f'Frequencies not found in {self.logfile}')
+                unscaled_frequencies = frequencies
+                vibration = HarmonicOscillator(frequencies=(frequencies, "cm^-1"))
+                # modes.append(vibration)
```
Comment (kspieks) [inline arkane/ess/orca.py:224]: Can you remove this line since it's not being used?

**Review by kspieks — COMMENTED** [review]
Thanks for adding an Orca adapter. Looks good and the tests are passing so this can probably be merged today. Can you clean up a few minor things that I left comments on?

File: arkane/ess/orca.py [hunk 9]
```diff
@@ -132,29 +169,86 @@ def load_geometry(self):
...
+        if len(inertia):
+            if inertia[0] == 0.0:
+                # If the first eigenvalue is 0, the rotor is linear
+                inertia.remove(0.0)
+                logging.debug('inertia is {}'.format(str(inertia)))
+                rotation = LinearRotor(inertia=(inertia, "amu*angstrom^2"), symmetry=symmetry)
+                rot.append(rotation)
+            else:
+                rotation = NonlinearRotor(inertia=(inertia, "amu*angstrom^2"), symmetry=symmetry)
+                # modes.append(rotation)
```
Comment (kspieks) [inline arkane/ess/orca.py:241]: Can you remove this line since it's not being used?

**Comment by dranasinghe** [comment]
In the case of new composite methods, only the energy need to be parse. I will add frequency scaling and BAC to rmg database.

File: rmgpy/tools/data/sim/mbSampled/simulate.log [hunk 10]
```diff
@@ -0,0 +1 @@
+Error: library reaction i2 <=> i1 is pressure dependent, and its high-pressure limit rate could not be extracted. Ignoring this library reaction (the rate could still be estimated from families).
```
Comment (kspieks) [inline rmgpy/tools/data/sim/mbSampled/simulate.log:1]: Is this log file needed? It looks like the simple example and liquid example only have `chem.inp`, `input.py`, and `species_dictionary.txt`, but maybe this is used somewhere and I just didn't notice

**Review by kspieks — COMMENTED** [review]
Thanks for adding these. Unit tests look good. Before merging, can you rebase and address the minor comments? For the LGTM, it could be confusing later on for the nested for loops to overwrite the same variable so can you change one of the `line` variable names in orca's load_conformer? Otherwise looks good

**Comment by alongd** [comment]
Merged as #2257

**Pull request closed without merging**